src/server/game/Spells/Spell.cpp

```diff
@@ -2287,6 +2287,10 @@ void Spell::DoAllEffectOnTarget(TargetInfo* target)
     if (!caster)
         return;
 
+    // Check for SPELL_ATTR7_INTERRUPT_ONLY_NONPLAYER before checking the miss conditions
+    if (m_spellInfo->HasAttribute(SPELL_ATTR7_INTERRUPT_ONLY_NONPLAYER) && unit->GetTypeId() != TYPEID_PLAYER)
```

Comment:
just a note, you probably should place if AFTER handling effects - the blizzlike effect is that if the stun interrupts the cast (if not immune), it should happen before this spell and not show in combat log as being interrupted by 32747